Documentation/fully-coupled-mesh-tutorial.rst

```diff
@@ -0,0 +1,127 @@
+Generating a CAD model for a fully-coupled earthquake-tsunami simulation
+==========================================================================
+
+Here we present a workflow for generating a CAD model of the Eastern Aegean Sea, including ocean and topography to be used in fully-coupled earthquake-tsunami simulations of the 2020 Aegean Sea earthquake.
+We use scripts from https://github.com/SeisSol/Meshing. To best follow this tutorial, we suggest adding the geometry script folder to the path:
+
+.. code-block:: bash
+
+    export PATH=$PATH:~/SeisSol/Meshing/creating_geometric_models
+
+Please consider running `git pull` in SeisSol/Meshing to pull the latest version of the scripts, if the repository is not newly cloned.
+You may also install required python modules with `pip3 install -r requirements.txt` (or if you use anaconda, `conda install --file requirements.txt`).
+This tutorial has been tested with SimModeler10.0-211122.
+Note that one specific step requires SimModeler11.0-220403-dev, but another step of the workflow fails with the dev version.
+
+Creating topography and boxes
+-------------------------------
+
+First, we download topography and bathymetry data from GEBCO
+(`http://www.gebco.net/ <http://www.gebco.net/>`__), and we triangulate it into a GoCad ts file.
+
+- Note that we downsample the topography data by a factor 2 for dealing with a reasonable size dataset in this tutorial.
+- Note also that we use a custom transverse Mercator roughly centered at the domain center.
+- With the option ``--change_zero_elevation 1.0``, we move the nodes with zero elevation to 1.0 m. This avoids having to intersect locally coplanar surfaces.
+- With the option ``--smooth 100``, we smooth the topography data which fall in the range :math:`\pm` 100 m. This facilitates the intersection of the sea surface with the topography and allows a smoother coastline (which can else have a saw-tooth shape due to rounded elevation data, stored as integers in Gebco files).
```

Comment:
What does it mean that the topography data that fall in the range of +/- 100 is smoothed?
Is the 100 a cutoff parameter or similar?

Reply:
Here is the code:
```python
    def smooth(self, zrange):
        "Smooth bathymetry in range +- zrange"
        "This is useful when preprocessing bathymetry data before intersection"
        import scipy.ndimage as ndimage

        z_smooth = ndimage.gaussian_filter(self.z, sigma=(1, 1), order=0)
        ids = np.where(abs(self.z) < zrange)
        self.z[ids] = z_smooth[ids]
```
The idea is to smooth the bathymetry (that is also degrade bathymetry data) only where it is necessary, that is at points with z close to 0, where the coastline should be.

Reply:
What about:
With the option ``--smooth 100``, we replace the elevation of topography nodes with z coordinates in the range :math:`\pm` 100 m by spatially smoothed (using a 2D Gaussian kernel) values.
This facilitates...

Reply:
Yup, sounds great. What happens with values > 100? They are not smoothed?

Reply:
no, they are not smoothed. The true topography data are not smooth. But smoothed data facilitate union (that's my experience at least).